bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDTransposeFlowGraphFinder.java

```diff
@@ -81,13 +85,49 @@ public List<? extends AbstractTransposeFlowGraph> findTransposeFlowGraphs(List<?
      * @param inputPins Relevant input pins on the given vertex
      * @return List of sinks created from the initial sink with previous vertices calculated
      */
-    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> inputPins, List<Node> sourceNodes) {
+    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> pins, List<Node> sourceNodes) {
         List<DFDVertex> vertices = new ArrayList<>();
         vertices.add(sink);
 
         if (sourceNodes.contains(sink.getReferencedElement())) {
             return vertices;
         }
+        
+        var inputPins = new ArrayList<>(pins);
+        
+        Map<Pin, List<Pin>> inToPreviousNodeInPinsMap = new HashMap<>();
+        inputPins.forEach(pin -> {
+        	Set<Pin> outputPins = new HashSet<>();
+        	inToPreviousNodeInPinsMap.put(pin, new ArrayList<>());
+        	dataFlowDiagram.getFlows()
+            .stream()
+            .filter(flow -> flow.getDestinationPin()
+                    .equals(pin))
+            .forEach(flow -> {
+            	outputPins.add(flow.getSourcePin());
+            });
+        	outputPins.stream().forEach(outPin -> {
+        		Behaviour behaviour = (Behaviour) outPin.eContainer();
+        		behaviour.getAssignment().stream().filter(it -> it.getOutputPin().equals(outPin)).forEach(it -> inToPreviousNodeInPinsMap.get(pin).addAll(it.getInputPins()));
+        	});
+        });
+                
+        Map<Pin, List<Pin>> mapInPinToEqualInPin = new HashMap<>();
+        var keyList = inToPreviousNodeInPinsMap.keySet().stream().toList();
+        for (int i = 0; i < keyList.size(); i++) {
+        	var key = keyList.get(i);
+        	var inPins = inToPreviousNodeInPinsMap.get(key);
+        	for (int j = i + 1; j < keyList.size(); j++) {
+        		var key2 = keyList.get(j);
+        		var inPin2 = inToPreviousNodeInPinsMap.get(key2);
+        		if (inPins.containsAll(inPin2) && inPin2.containsAll(inPins)) {
+        			if (mapInPinToEqualInPin.getOrDefault(key, null) == null) mapInPinToEqualInPin.put(key, new ArrayList<>());
+        			mapInPinToEqualInPin.get(key).add(key2);
+        		};
+        	}
+        }
```

Comment:
i guess this is a function for PCM specific stuff, like calling and returning right?

Reply:
This is to avoid long copy and unification chains when there are multiple flows from one node to another, as we often see when transforming from PCM. This is only done when the outPins depend on the same Input pins. This functions compares the required input pins for the output pins of the incoming flows and skips the unnecessary steps